src/common.rs

```diff
@@ -0,0 +1,173 @@
+//! Common functionality you might need when using Specs.
+//!
+//! At the moment, this module provides two types:
+//!
+//! * `Errors`: A resource you can use to store errors that occurred outside of
+//!             the ECS but were catched inside, therefore should be handled by the user
+//!
+//! * `Merge`: A system generic over `T` which automatically merges `Ready` futures into
+//!            the component storage for `T`.
+//!
+//! To make use of these features, you need to ask for the `common` feature
+//! like this:
+//!
+//! ```toml
+//! [dependencies.specs]
+//! # version = "..."
+//! features = ["common"]
+//! ```
+
+use std::convert::AsRef;
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
+use std::io::Write;
+use std::marker::PhantomData;
+
+use futures::{Async, Future};
+use {Component, DenseVecStorage, Entities, FetchMut, Join, RunningTime, System, WriteStorage};
+
+/// A boxed error implementing `Debug`, `Display` and `Error`.
+pub struct BoxedErr(pub Box<Error + Send + Sync + 'static>);
+
+impl BoxedErr {
+    /// Creates a new boxed error.
+    pub fn new<T>(err: T) -> Self
+        where T: Error + Send + Sync + 'static
+    {
+        BoxedErr(Box::new(err))
+    }
+}
+
+impl AsRef<Error> for BoxedErr {
+    fn as_ref(&self) -> &(Error + 'static) {
+        self.0.as_ref()
+    }
+}
+
+impl Debug for BoxedErr {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        write!(f, "{:?}", self.as_ref())
+    }
+}
+
+impl Display for BoxedErr {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        write!(f, "{}", self.as_ref())
+    }
+}
+
+impl Error for BoxedErr {
+    fn description(&self) -> &str {
+        self.as_ref().description()
+    }
+}
+
+/// The component type for futures of `T`.
+pub type FutureComponent<T> = Box<Future<Item = T, Error = BoxedErr> + Send + Sync>;
```

Comment:
Well if you just use trait bounds then the user gets to choose do they want to box it. The point is that if you know that certain component comes only from certain future there is no need for boxing.